partner_preview_scripts/ValidationLab.py

```diff
@@ -3,432 +3,472 @@
+            if biomarker in biomarker_df.columns
+        ]
 
-        # The first column is the gene name, the rest are the hypotheses:
-        hypothesis_columns = hypothesis_df.columns[1:]
+        # Applying the full map on the dataframe one-by-one:
+        processed_biomarkers = reduce(
+            lambda DF, value: DF.withColumn(*value),
+            map(
+                # Function to process biomarker:
+                lambda biomarker: (
+                    biomarker,
+                    cls._get_biomarker_wrapper(cls.BIOMARKERMAPS)(
+                        f.lit(biomarker),
+                        f.col(biomarker),
+                    ),
+                ),
+                # Iterator to apply the function over:
+                biomarkers_in_data,
+            ),
+            biomarker_df,
+        )
 
-        unpivot_expression = f"""stack({len(hypothesis_columns)}, {", ".join([f"'{x}', `{x}`" for x in hypothesis_columns])} ) as (hypothesis, {call})"""
+        # The biomarker columns are unstacked into one single 'biomarkers' column:
+        biomarker_unstack = f"""stack({len(biomarkers_in_data)}, {", ".join([f"'{x}', {x}" for x in biomarkers_in_data])}) as (biomarker_name, biomarker)"""
 
         return (
-            hypothesis_df.select("Gene", f.expr(unpivot_expression))
-            .withColumn(call, f.col(call).cast(t.BooleanType()))
+            processed_biomarkers
+            # Selecting cell line name, cell line annotation and applyting the stacking expression:
+            .select(
+                f.col("Cell Line Name").alias("cellLineName"),
+                f.expr(biomarker_unstack),
+            )
+            # Filter out all null biomarkers:
+            .filter((f.col("biomarker").isNotNull()))
+            # Grouping data by cell lines again:
+            .groupBy("cellLineName")
+            .agg(
+                f.collect_list("biomarker").alias("biomarkerList"),
+            )
             .persist()
         )
 
 
-@f.udf(
-    t.StructType(
-        [
-            t.StructField("name", t.StringType(), False),
-            t.StructField("description", t.StringType(), False),
-        ]
-    )
-)
-def get_biomarker(column_name, biomarker):
-    """This function returns with a struct with the biomarker name and description"""
-
-    # If the biomarker has a direct mapping:
-    if "direct_mapping" in BIOMARKERMAPS[column_name]:
-        try:
-            return BIOMARKERMAPS[column_name]["direct_mapping"][biomarker]
-        except KeyError:
-            logging.warning(
-                f"Could not find direct mapping for {column_name}:{biomarker}"
+@dataclass
+class ValidationLabEvidenceParser:
+    projects: list
+    assays: list
+    sharedParemeters: dict
+
+    @staticmethod
+    def _get_disease_cell_lines(model_passport_df: DataFrame) -> DataFrame:
+        return (
+            # The following option is required to correctly parse CSV records which contain newline characters.
```

Comment:
This comment doesn't belong here
```suggestion
```

Reply:
Yes, indeed.

Reply:
I'm dropping this function, also quite artificial unit of logic